pennylane/io/qasm_interpreter.py

```diff
@@ -946,19 +1091,21 @@ def visit_cast(self, node: ast.Cast, context: Context):
             TypeError: If the cast cannot be made.
         """
         arg = self.visit(node.argument, context)
-        ret = arg
         try:
-            if isinstance(node.type, ast.IntType):
-                ret = int(arg)
-            if isinstance(node.type, ast.UintType):
-                ret = uint(arg)
-            if isinstance(node.type, ast.FloatType):
-                ret = float(arg)
-            if isinstance(node.type, ast.ComplexType):
-                ret = complex(arg)
-            if isinstance(node.type, ast.BoolType):
-                ret = bool(arg)
-            # TODO: durations, angles, etc.
+            match node.type.__class__:
+                case ast.IntType:
+                    ret = int(arg)
+                case ast.UintType:
+                    ret = uint(arg)
+                case ast.FloatType:
+                    ret = float(arg)
+                case ast.ComplexType:
+                    ret = complex(arg)
+                case ast.BoolType:
+                    ret = bool(arg)
```

Comment:
Hmm, this doesn't work?
```suggestion
            match node:
                case ast.IntType():
                    ret = int(arg)
                case ast.UintType():
                    ret = uint(arg)
                case ast.FloatType():
                    ret = float(arg)
                case ast.ComplexType():
                    ret = complex(arg)
                case ast.BoolType():
                    ret = bool(arg)
```

Reply:
The `node` is an instance of a class like `BoolType` which has attributes such as a `span` object that indicates the line number that the cast occurs on in the QASM. I am not interested in checking that this instance is equal to another instance of `BoolType` (for example) that has different attributes. What we are trying to check here is that the class of the `node.type` object is `BoolType` or `ComplexType` etc. and I would like the code to reflect that intention clearly.

Reply:
IMHO dunders are great! So handy.

Reply:
If it's ok, since this is a very minor stylistic preference, and we have a tight timeline, I will merge it.